### PR TITLE
Support msrc types

### LINKED ISSUE #381: msrc package type
**What would you like to be added**: Package Type support for MSRC

**Additional context**:
Probably just the types so that they can be filled in by injected sbom. No direct support for scanning/parsing to provide MSRC package type (which is basically KBs in reality)

### PR DESCRIPTION
**Pull request by alfredodeza**
Closes #381

### CODE REVIEW

File: syft/distro/identify_test.go [hunk 1]
```diff
@@ -88,6 +89,11 @@ func TestIdentifyDistro(t *testing.T) {
 
 	observedDistros := internal.NewStringSet()
 	definedDistros := internal.NewStringSet()
+
+	// Somewhat cheating with Windows. There is no support for detecting/parsing a Windows OS, so it is not
+	// possible to comply with this test unless it is added manually to the "observed distros"
+	observedDistros.Add(string(Windows))
```
Comment (wagoodman) [inline syft/distro/identify_test.go:95]: agreed on not requiring testing of windows, however, I feel that removing the windows distro type from the `definedDistros` variable is a bit better than saying we observed windows during testing.

**Review by wagoodman — CHANGES_REQUESTED** [review]
I see that there is a need to capture KB metadata, but does that need to be done within `pkg`? That is, adding a new package type and package metadata seems to imply that this is package metadata, when this is really more akin to distro metadata.

File: syft/distro/identify_test.go (hunk 1 above)
Comment (alfredodeza) [reply syft/distro/identify_test.go:95]: I see those as the same thing but I'm fine swapping it around

**Comment by alfredodeza** [comment]
> I see that there is a need to capture KB metadata, but does that need to be done within `pkg`? That is, adding a new package type and package metadata seems to imply that this is related to a package, when this is really more akin to distro metadata.

The "packages" are really distro information. There aren't any packages here, I'm overloading the package to represent each KB installed in a Windows OS. The metadata is not (in my testing) used, but it represents better what the fields should look like, and opens up the ability to add more metadata later.

File: syft/distro/identify_test.go (hunk 1 above)
Comment (alfredodeza) [reply syft/distro/identify_test.go]: This got swapped around

**Review by wagoodman — COMMENTED** [review]
Thanks for the offline conversation regarding this approach and why it plays nicely into the existing grype patterns. Though it would be nice to capture this under `distro` as auxiliary metadata that can get picked up by a custom grype matcher, currently the matchers are oriented around matching packages... and even though they are passed distro information, we would need to add an alternate matching pattern that could take extra time. 

I think you're right --we can get away with not doing that work for now until we find other examples where a similar matching pattern would be needed, then take the effort to refactor. 

(you can ignore my comment here https://github.com/anchore/syft/pull/389#pullrequestreview-637661558)

**Review by wagoodman — APPROVED** [review]

**Pull request merged**